### PR TITLE
Timeline optimizations

### PR DESCRIPTION
**Pull request by dmitrio95**
As full timeline refactoring (#4542) has not been finished as for this moment, this PR tries to fix the most prominent performance issues of the old Timeline implementation. In general, the following optimizations are implemented:
1) Optimize the procedure of highlighting currently visible measures by avoiding iterating over all timeline items. This mostly affects score scrolling which could previously be noticeably slowed down on large scores if Timeline was open.
2) Implement partial updating of measure rectangles in Timeline. Top panel with metadata (tempo changes, time signatures etc.) is still fully updated on each action but the overall procedure of Timeline updating is still much faster than it was before.
3) Don't update data of invisible Timeline when loading or switching a score. This case has been missed from the previous optimizations for not updating invisible Timeline.

Although Timeline performance is still not ideal with this PR and could be optimized further, this PR should still hopefully make Timeline more usable for people working on relatively large scores. Users who don't use Timeline should also benefit from this PR due to the last optimization for invisible Timeline.

### CODE REVIEW

**Comment by Howard-C** [comment]
Great! What about the inability to cooperate with dark mode?

File: mscore/timeline.cpp [hunk 1]
```diff
@@ -2247,18 +2328,55 @@ void Timeline::updateGrid()
...
+      const CmdState& state = _score->cmdState();
+
+      const bool layoutChanged = state.layoutRange();
+
+      if (!layoutChanged) {
+            updateGridView();
+            return;
+            }
+
+      const bool layoutAll = layoutChanged && (state.startTick().negative() || state.endTick().negative());
```
Comment (Jojo-Schmitz) [inline mscore/timeline.cpp]: See https://travis-ci.org/musescore/MuseScore/jobs/652005292#L2630-L2636, "`class Ms::Fraction` has no member named `negative`"

Comment (dmitrio95) [reply mscore/timeline.cpp]: Looks like I forgot to commit some changes, as always. Will update PR soon.

**Comment by dmitrio95** [comment]
> What about the inability to cooperate with dark mode?

This pull request doesn't address that, it is solely about performance of using Timeline.

File: mscore/timeline.cpp (hunk 1 above)
Comment (AlbertoEAF) [reply mscore/timeline.cpp]: Any update on this?

**Pull request merged**